assets/questions/unknownAlphabet.js

```diff
@@ -0,0 +1,326 @@
+      [
+        'Next, double-check your code to make sure it handles unexpected cases',
+        ', such as a test case with no solution or a test case with more than ',
+        'one correct answer.'
+      ].join(''),
+      [
+        'For the test case that has no solution, just return "". For the test ',
+        'case that has multiple answers, return the first answer in English ',
+        'lexicographic (alphabetical) order.'
+      ].join(''),
+      'Example dictionary:',
+      '[a, b, a]',
+      'Alphabet is: ""',
+      '[cba]',
+      'Alphabet is: "abc"'
+    ],
+    prerequisiteSkills: ['Topological Sorting', 'String Manipulation'],
+    acquiredSkills: ['String Manipulation'],
+    inputFunctionName: null,
+    outputFunctionName: null,
+    mainFunctionName: 'findDictionary',
+    correctnessTests: [{
+      input: [
+        "a",
+        "b",
+        "a"
+      ],
+      allowedOutputs: ['']
+    }, {
+      input: [
+        "c",
+        "a",
+        "b",
+        "a"
+      ],
+      allowedOutputs: ['']
+    }, {
+      input: [
+        "cab",
+        "bc"
+      ],
+      allowedOutputs: ['acb']
+    }],
+    buggyOutputTests: [{
+      buggyFunctionName: 'AuxiliaryCode.failsOnNoSolution',
+      messages: [
+        [
+          'What if the test case is ["c", "a", "b", "a"]?',
+          'And what answer your code will generate for this test case?'
+        ].join(''),
+        [
+          'The answer to that case should be a empty string. But what your code',
+          ' have returned?'
+        ].join('')
+      ]
+    }],
+    performanceTests: []
+  }, {
+    instructions: [
+      [
+        'Finally, modify your code to handle strings with a mix of lowercase ',
+        'and uppercase letters. Assume that a lowercase and uppercase version ',
+        'of a letter is the same with regards to lexicographic ordering. Your ',
+        'answer, though, should be lowercase.'
+      ].join(''),
+      'Example dictionary:',
+      '[b, Ba]',
+      'Alphabet is: "ab"'
+    ],
+    prerequisiteSkills: ['Topological Sorting', 'String Manipulation'],
+    acquiredSkills: ['String Manipulation'],
+    inputFunctionName: null,
+    outputFunctionName: null,
+    mainFunctionName: 'findDictionary',
+    correctnessTests: [{
+      input: [
+        "ab",
+        "Aba",
+        "b"
+      ],
+      allowedOutputs: ['ab']
+    }],
+    buggyOutputTests: [{
+      buggyFunctionName: 'AuxiliaryCode.failsOnUppercase',
+      messages: [
+        [
+          'Try running your code over ["a", "Ab", "b"]. Do you get the result ',
+          'you expected?'
+        ].join(''),
+        'How does your code handle uppercase letters?',
+        [
+          'The answer to the test case ["a", "Ab", "b"] should be "ab". Did ',
+          'your code return the string with some letters in uppercase?'
+        ].join('')
+      ]
+    }],
+    // TODO: Time complexity is O(m), which is non-linear. Fix in the future.
```

Comment:
And what's more, it's hard to increase the size of the input_data_atom only based on the input data atom